src/components/PopupWithForm.js

```diff
@@ -8,20 +8,31 @@ export default class PopupWithForm extends Popup {
       this._popupForm.querySelectorAll(".modal__input")
     );
     this._handleFormSubmit = handleFormSubmit;
+    this._modalSubmitButton =
+      this._popupElement.querySelector(".modal__button");
   }
 
-  _setInputValues() {
-    this._formValues = {};
+  _getInputValues() {
+    const formValues = {};
     this._inputValues.forEach((input) => {
-      this._formValues[input.name] = input.value;
+      formValues[input.name] = input.value;
     });
-    return this._formValues;
+    return formValues;
   }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._modalSubmitButton.textContent = "Saving...";
+    } else {
+      this._modalSubmitButton.textContent = "Save";
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`